plugins/isp/internode_usage

```diff
@@ -83,31 +83,25 @@ fi
 
 xpath_extract() {
 	xpath="$1"
-	# use trap to output an error message but let the caller get the status
-	# shellcheck disable=SC2064
-	trap "echo error extracting ${xpath} >&2" EXIT
 	xpath -q -n -e "${xpath}" \
-		| sed 's/<\([^>]*\)>\([^<]*\)<[^>]*>/\2/;s^N/A^U^'
-	trap - EXIT
+		| sed 's/<\([^>]*\)>\([^<]*\)<[^>]*>/\2/;s^N/A^U^' \
+		|| (echo "error extracting ${xpath}" >&2 \
+		    && false)
 }
 
 xpath_extract_attribute() {
 	xpath="$1"
-	# use trap to output an error message but let the caller get the status
-	# shellcheck disable=SC2064
-	trap "echo error extracting attribute at ${xpath} >&2" EXIT
-	xpath -q -n -e "${xpath}" \
-		| sed 's/.*="\([^"]\+\)".*/\1/'
-	trap - EXIT
+	(xpath -q -n -e "${xpath}" \
+		| sed 's/.*="\([^"]\+\)".*/\1/') \
```

Comment:
I think, here (and above in the other function) you merely test, wether `sed` failed before.

I guess, you will need to store the output of `xpath` in a variable :)

Reply:
Ah, you're right...

This is annoying, as I was trying to keep it short and sweet \: